### DISCUSSION TITLE
Running out of TCP Connections

### DISCUSSION THREAD

**Discussion by trademark18** [artilleryio/artillery#1037] [Artillery]
Hi all, 

I'm new to Artillery, and full disclosure: I'm actually using a [fork ](https://github.com/vadivelselvaraj/artillery)(it supports variable substitution _and_ the [artillery-plugin-aws-sigv4](https://github.com/Nordstrom/artillery-plugin-aws-sigv4) at the same time -- see #544)

Anyway, I already ran into the issue where you run out of open files on WSL2 (Ubuntu) and I found [somewhere ](https://github.com/microsoft/WSL/issues/4575#issuecomment-709586759) that you can fix that by doing this:
```bash
mylimit=65535
sudo prlimit --nofile=$mylimit --pid $$; ulimit -n $mylimit
```

My new problem is similar.  This time I'm running out of TCP connections.  From what I read I can increase the connections like so:
```bash
 sudo sysctl -w net.ipv4.ip_local_port_range="32768 65535"
```
I can't seem to raise it above 65535, and the original (where I was running out) was 61000 so that's not much more room.

So -- the questions for discussion:
- **How can I work around the limitation of TCP connections?**
- **Is there a way to make artillery close/release TCP connections it's done with?**

Thanks for the work on this really helpful tool!

P.S. Let me know if you want this to be an issue instead.

**Comment by hassy** [comment]
Artillery does close TCP connections once a virtual user that uses that connection is finished. Do your tests have long-running scenarios to cause the machine to run out of ports?

In theory it's impossible to open more than 65k connections from a single client to a server, you will need to add more machines running Artillery to go above the limit.

**Reply by trademark18** [reply]
The further I got researching this the more it looked like Artillery was probably not the issue.  Honestly I'm struggling to remember exactly what it was, but I know some issues I had were related to the API I was testing against, so it could have been that.  Thanks for the reply!
